src/include/daos/tests_lib.h

```diff
@@ -42,6 +42,12 @@
 	if (__rc != 0) \
 		fail_msg("Not successful!! Error code: " DF_RC, DP_RC(__rc)); \
 	} while (0)
+#define assert_rc_equal(rc, expected_rc) do { \
+	if((rc) == (expected_rc)) \
+		break; \
+	assert_string_equal(d_errstr(rc), d_errstr(expected_rc)); \
```

Comment:
I would add a print_message above this line to print the numeric codes just in case.   Some APIs will return a positive value or 0 on success so will be translated as DER_UNKNOWN